packages/grid/x-grid/src/pagination.XGrid.test.tsx

```diff
@@ -0,0 +1,57 @@
+import { render } from '@testing-library/react';
+import * as React from 'react';
+import { raf, sleep } from 'test/utils/helperFn';
+import { expect } from 'chai';
+import { XGrid, useApiRef } from '@material-ui/x-grid';
+
+describe('<XGrid />', () => {
+  const baselineProps = {
+    rows: [
+      {
+        id: 0,
+        brand: 'Nike',
+      },
+      {
+        id: 1,
+        brand: 'Adidas',
+      },
+      {
+        id: 2,
+        brand: 'Puma',
+      },
+    ],
+    columns: [{ field: 'brand', width: 100 }],
+  };
+
+  before(function beforeHook() {
+    if (/jsdom/.test(window.navigator.userAgent)) {
+      // Need layouting
+      this.skip();
+    }
+  });
+
+  describe('Pagination', () => {
+    it('should apply setPage correctly', async () => {
+      let apiRef;
+      const GridTest = () => {
+        apiRef = useApiRef();
+
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid {...baselineProps} apiRef={apiRef} pagination pageSize={1} hideFooter />
+          </div>
+        );
+      };
+
+      render(<GridTest />);
+
+      let cell = document.querySelector('[role="cell"][aria-colindex="0"]')!;
+      expect(cell).to.have.text('Nike');
+      apiRef.current.setPage(2);
+      await sleep(100);
```

Comment:
Regarding the raf -> sleep. I wonder why we had a raf in the first place. It's unusual, is there a settimout/raf in the codebase that requires it? It would be interesting to dive into it, it could hide something.

Reply:
There 2 renderings going on, and setState is async in react.

Reply:
Oh so we need to wrap `apiRef.current.setPage(2);` with act().

Reply:
This could make a good follow-up pull request

```suggestion
import { act } from 'test/utils';

act(() => {
  apiRef.current.setPage(2);
});
```